articles/articles/article.php

```diff
@@ -1,176 +1,338 @@
+				$art_obj->setState('filter.end_date_range', $end_date);
 			}
 		}
-		
-		//get article data
-		/*if($article_id)
-		{
-			$artcl = new ContentModelArticle();
-			$adetails = $artcl->getItem($article_id);
-			//$art_obj->setState('filter.article_id',$article_id );
-		}*/
 
 		$rows = $art_obj->getItems();
 
-		//test code
-		/*JPluginHelper::importPlugin( 'content' );
-		$dispatcher = JEventDispatcher::getInstance();
-		$results = $dispatcher->trigger( 'onContentPrepare', array( 'com_content', &$row , ) );
-		*/
-		//
+		$num_articles = $art_obj->getTotal();
 
-		//format data
-		$obj = new BlogappSimpleSchema();
-		//$dispatcher = JEventDispatcher::getInstance();
-		//$item = $obj->mapPost($a_data,'', 100, array('text'));
-		foreach($rows as $row)
+		foreach ($rows as $subKey => $subArray)
 		{
-			if(!isset($row->text)||empty($row->text))
+			if ($subArray->catid)
+			{
+				$subArray->catid = array('catid' => $subArray->catid,'title' => $subArray->category_title);
+			}
+
+			if ($subArray->created_by)
 			{
-				$row->text = $row->introtext." ".$row->fulltext;
+				$subArray->created_by = array('id' => $subArray->created_by,'name' => $subArray->author,
+				'avatar' => "http://172.132.45.45/joomla/investsure/images/samlpe.jpg");
 			}
-			/*
-			$document	= JFactory::getDocument();
-			$document->setType('html');
-
-			JPluginHelper::importPlugin( 'content' );
-			$dispatcher = JEventDispatcher::getInstance();
-			$dispatcher->trigger('onContentPrepare', array ('com_content.article', &$row, &$row->params, null));
-			*/
-			
-			$item = $obj->mapPost($row,'', 100, array());
-		}
-	
-		$this->plugin->setResponse($item);
-		
+
+			if ($subArray->images)
+			{
+				$images = json_decode($subArray->images);
+
+				foreach ($images as $key => $value)
+				{
+					if ($value)
+					{
+						$images->$key = JURI::base() . $value;
+					}
+				}
+
+				$subArray->images = $images;
+			}
+
+			unset($subArray->checked_out);
+			unset($subArray->checked_out_time);
+			unset($subArray->created_by_alias);
+			unset($subArray->modified_by);
+			unset($subArray->modified_by_name);
+			unset($subArray->urls);
+			unset($subArray->attribs);
+			unset($subArray->metadata);
+			unset($subArray->metakey);
+			unset($subArray->metadesc);
+			unset($subArray->xreference);
+			unset($subArray->readmore);
+			unset($subArray->category_route);
+			unset($subArray->category_access);
+			unset($subArray->category_alias);
+			unset($subArray->author_email);
+			unset($subArray->parent_title);
+			unset($subArray->parent_id);
+			unset($subArray->parent_route);
+			unset($subArray->parent_alias);
+			unset($subArray->rating);
+			unset($subArray->rating_count);
+			unset($subArray->published);
+			unset($subArray->parents_published);
+			unset($subArray->alternative_readmore);
+			unset($subArray->layout);
+			unset($subArray->params);
+			unset($subArray->displayDate);
+			unset($subArray->author);
+			unset($subArray->category_title);
+		}
```

Comment:
@vijaykhollam we discussed that instead of unsetting variables that are not needed, just set the ones that you want to use.